app/controllers/legacy_routes/handler.rb

```diff
@@ -5,13 +5,13 @@
 #  See the COPYING file at the top-level directory or at
 #  https://github.com/puzzle/cryptopus.
 
-class LegacyRoutes::Locale
+class LegacyRoutes::Handler
 
   LOCALES_REGEX = I18n.available_locales.map do |locale|
     "\/#{locale}|"
   end.join('').chop.freeze
 
   def redirect_url(url)
-    url.downcase.remove(/#{LOCALES_REGEX}/)
+    url.downcase.remove(/#{LOCALES_REGEX}/).sub(/(accounts|groups)$/, '\2')
```

Comment:
def redirect_to
   remove_locale
end

use ruby case ... https://www.rubyguides.com/2015/10/ruby-case/

when legacy_accounts_path?
   account_path(account_id)

\# ../teams/42/groups/33/accounts/45 -> /accounts/45 
def legacy_accounts_path?

add include Rails.application.routes.url_helpers to get all path/url helpers in controller